flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/history/HistoryServerTest.java

```diff
@@ -117,45 +124,145 @@ public void testHistoryServerIntegration() throws Exception {
 		for (int x = 0; x < numJobs; x++) {
 			runJob();
 		}
+		final int numLegacyJobs = 1;
 		createLegacyArchive(jmDirectory.toPath());
 
-		CountDownLatch numExpectedArchivedJobs = new CountDownLatch(numJobs + 1);
+		waitForArchivesCreation(numJobs + numLegacyJobs);
 
-		Configuration historyServerConfig = new Configuration();
-		historyServerConfig.setString(HistoryServerOptions.HISTORY_SERVER_ARCHIVE_DIRS, jmDirectory.toURI().toString());
-		historyServerConfig.setString(HistoryServerOptions.HISTORY_SERVER_WEB_DIR, hsDirectory.getAbsolutePath());
-		historyServerConfig.setLong(HistoryServerOptions.HISTORY_SERVER_ARCHIVE_REFRESH_INTERVAL, 100L);
+		CountDownLatch numExpectedArchivedJobs = new CountDownLatch(numJobs + numLegacyJobs);
 
-		historyServerConfig.setInteger(HistoryServerOptions.HISTORY_SERVER_WEB_PORT, 0);
+		Configuration historyServerConfig = createTestConfiguration(false);
 
-		// the job is archived asynchronously after env.execute() returns
-		File[] archives = jmDirectory.listFiles();
-		while (archives == null || archives.length != numJobs + 1) {
-			Thread.sleep(50);
-			archives = jmDirectory.listFiles();
+		HistoryServer hs =
+		new HistoryServer(
+			historyServerConfig,
+			(event) -> {
+				if (event.getType() == HistoryServerArchiveFetcher.ArchiveEventType.CREATED) {
+					numExpectedArchivedJobs.countDown();
+				}
+			});
+
+		try {
+			hs.start();
+			String baseUrl = "http://localhost:" + hs.getWebPort();
+			numExpectedArchivedJobs.await(10L, TimeUnit.SECONDS);
+
+			ObjectMapper mapper = new ObjectMapper();
+			Assert.assertEquals(numJobs + numLegacyJobs, getJobsOverview(baseUrl, mapper).getJobs().size());
+
+		} finally {
+			hs.stop();
+		}
+	}
+
+	@Test
+	public void testCleanExpiredJob() throws Exception {
+		runArchiveExpirationTest(true);
+	}
+
+	@Test
+	public void testRemainExpiredJob() throws Exception {
+		runArchiveExpirationTest(false);
+	}
+
+	private void runArchiveExpirationTest(boolean cleanupExpiredJobs) throws Exception {
+		int numExpiredJobs = cleanupExpiredJobs ? 1 : 0;
+		int numJobs = 3;
+		List<String> jobIDs = new ArrayList<>();
```

Comment:
```suggestion
		List<String> jobIds = new ArrayList<>();
```
This is a more common capitalization for ID, at least in newer code.